### ISSUE TITLE
Suggestion: execute fetch requests in browser when in Dev

### ISSUE THREAD

**Issue by yasserlens** [vendure-ecommerce/storefront-qwik-starter#110]
I found debugging API fetches hard because of 2 things:
1) queries execute on the server in utils/api.ts (execute(..)). This makes using the GraphQL debugging chrome extension impossible. 
2) the main issue was that errors weren't returned from the function utils/api.ts -> extractTokenAndData(..). We need to at least log errors out in dev, and probably expose errors to the client so the app knows what to do when data is null.

For #1, I did the following inside the execute function, which decides where the call goes (server or client)
```
const response: ResponseProps<R> = isBrowser && !import.meta.env.DEV
		? await executeOnTheServer(requestOptions, options.apiUrl!)
		: await executeRequest(requestOptions, options.apiUrl!);
```

For #2 I'm trying to find the best way to handle this. Exposing errors to the app should be safe, for the app to translate what the error should mean to the end user. I just have to find a way to get this data to my components/routes. If you have ideas let me know.

**Comment by gioboa** [comment]
Thanks for your feedback. 
1) this is a nice improvement, pls open a PR for that
2) which kind of error need to be exposed to the end user?
If we have a status code we can translate it in various languages

**Comment by yasserlens** [comment]
1) Done - https://github.com/vendure-ecommerce/storefront-qwik-starter/pull/111
2) The final error message will depend on each store (depends on what the error is), but in general, a "An error occurred, could not do XYZ" would help even if only as console.log messages. In our case, it was the attempts to fetch entities. We'd like to solve this by returning the errors object from the response.json() output in api.ts.

**Comment by yasserlens** [comment]
I tried to fix #2 with the following pull request:
https://github.com/vendure-ecommerce/storefront-qwik-starter/pull/112

This works so far for me locally. Let me know if it doesn't make sense. I haven't tested this thoroughly.

**Comment by gioboa** [comment]
Thanks for your PR, it looks good
